### PR TITLE
feat: Add modal for transaction sign guidance

### PR DESCRIPTION
**Pull request by yigiterdev**
- Ticket: https://linear.app/hipo/issue/PWEB-18/connect-modal-for-transaction-sign-guidance

image

### CODE REVIEW

**Comment by mucahit** [comment]
I think we should have an option to disable the toast, it can be enabled by default. I'm pretty sure some of the dApps are handling the guidance on their own. 

What do you think? @yigiterdev

**Comment by yigiterdev** [comment]
> I think we should have an option to disable the toast, it can be enabled by default. I'm pretty sure some of the dApps are handling the guidance on their own.
> 
> What do you think? @yigiterdev

I added to `PeraWalletConnectOptions`. I'm not sure this is the best way, but here's how to use it.

`const peraWallet = new PeraWalletConnect({
  shouldShowSignTxnToast: false
});`

BTW, `shouldShowSignTxnToast`'s default is `true`. If that's okay, it might be good to add it to the readme.

File: src/modal/peraWalletConnectModalUtils.tsx [hunk 1]
```diff
@@ -9,13 +9,17 @@ import PeraWalletConnectModal from "./PeraWalletConnectModal";
...
+import PeraWalletSignTxnToast from "./sign-toast/PeraWalletSignTxnToast";
 
 // The ID of the wrapper element for PeraWalletConnectModal
 const PERA_WALLET_CONNECT_MODAL_ID = "pera-wallet-connect-modal-wrapper";
 
 // The ID of the wrapper element for PeraWalletRedirectModal
 const PERA_WALLET_REDIRECT_MODAL_ID = "pera-wallet-redirect-modal-wrapper";
 
+// The ID of the wrapper element for PeraWalletSignTxnToast
+const PERA_WALLET_SIGN_TXN_TOAST = "pera-wallet-sign-txn-toast";
```
Comment (mucahit) [inline src/modal/peraWalletConnectModalUtils.tsx:21]:
```suggestion
const PERA_WALLET_SIGN_TXN_TOAST_ID = "pera-wallet-sign-txn-toast";
```

File: src/modal/peraWalletConnectModalUtils.tsx [hunk 2]
```diff
@@ -82,6 +86,24 @@ function openPeraWalletRedirectModal() {
   }
 }
 
+/**
+ * Creates a PeraWalletRedirectModal instance and renders it on the DOM.
```
Comment (mucahit) [inline src/modal/peraWalletConnectModalUtils.tsx:90]: Let's update the JSDoc

File: src/modal/sign-toast/_pera-wallet-sign-txn-toast.scss [hunk 3]
```diff
@@ -0,0 +1,89 @@
+@import "../../ui/styles/media-queries";
+
+.pera-wallet-sign-txn-toast {
+  --pera-wallet-sign-txn-toast-width: 422px;
+  --pera-wallet-sign-txn-toast-height: 134px;
+  --pera-wallet-sign-txn-toast-font-family: "Inter", sans-serif;
+
+  position: absolute;
```
Comment (mucahit) [inline src/modal/sign-toast/_pera-wallet-sign-txn-toast.scss:8]: Shouldn't it be `position: fixed`?

**Review by mucahit — CHANGES_REQUESTED** [review]
thanks!! 🎉 

Let's also document `shouldShowSignTxnToast` option in the Readme

File: README.md [hunk 4]
```diff
@@ -138,6 +138,14 @@ try {
 }
 ```
 
+### Sign Transaction Toast
+
+`@perawallet/connect` informs users with toast when users try to sign transaction. If dApp wants to disable it, it can be disabled as follows. The default value of `shouldShowSignTxnToast` is `true`.
```
Comment (mucahit) [inline README.md:143]: Let's move this paragraph under the `Sign Transaction` title after the first paragraph.

---

I'd also change a bit the paragraph


`@perawallet/connect` guides users with a toast message when the `signTransaction` is triggered. It's enabled by default but in some cases, you may not need to the toast message (e.g. you already have signing guidance for users). There's a configuration called `shouldShowSignTxnToast` to disable it, see the example below

```js
const peraWallet = new PeraWalletConnect({shouldShowSignTxnToast: false});
```

File: src/PeraWalletConnect.ts [hunk 5]
```diff
@@ -59,6 +63,9 @@ class PeraWalletConnect {
     }
 
     this.connector = null;
+    this.shouldShowSignTxnToast = options?.shouldShowSignTxnToast
```
Comment (mucahit) [inline src/PeraWalletConnect.ts:66]: How is this one?
```suggestion
    this.shouldShowSignTxnToast = typeof options?.shouldShowSignTxnToast === "undefined" 
    ? true : 
    options.shouldShowSignTxnToast
```

**Review by mucahit — CHANGES_REQUESTED** [review]

**Review by mucahit — APPROVED** [review]

**Pull request merged**